rasa/nlu/test.py

```diff
@@ -654,28 +654,19 @@ def _calculate_report(
     if report_as_dict is None:
         report_as_dict = bool(output_directory)
 
-    if output_directory:
-        report, precision, f1, accuracy = get_evaluation_metrics(
-            targets,
-            predictions,
-            output_dict=report_as_dict,
-            exclude_label=exclude_label,
-        )
+    report, precision, f1, accuracy = get_evaluation_metrics(
+        targets, predictions, output_dict=report_as_dict, exclude_label=exclude_label,
+    )
+
+    if report_as_dict:
         report = _add_confused_labels_to_report(
             report,
             confusion_matrix,
             labels,
             exclude_labels=[exclude_label] if exclude_label else [],
         )
     else:
-        report, precision, f1, accuracy = get_evaluation_metrics(
-            targets,
-            predictions,
-            output_dict=report_as_dict,
-            exclude_label=exclude_label,
-        )
-        if isinstance(report, str):
-            log_evaluation_table(report, precision, f1, accuracy)
+        log_evaluation_table(report, precision, f1, accuracy)
```

Comment:
Why only log in this case?

Reply:
Whenever we store the report to disk it will be a formatted string report and `report_as_dict` will be `True` (from the CLI). Only if `report_as_dict` is `False` (either specified manually or not specified + no output directory provided), we get the formatted string report which needs to be logged. We could change this to `elif not output_directory` 🤔

Reply:
Changed this to the `elif` part btw.